controllers/testclustertopology_controller.go

```diff
@@ -260,6 +260,27 @@ func (r *TestClusterTopologyReconciler) installTiDBCluster(ctx context.Context,
 	return false, tiupCtl.InstallCluster(log, ct.Name, ct.Spec.TiDBCluster.Version)
 }
 
+func (r *TestClusterTopologyReconciler) upgradeTiDBCluster(ctx context.Context, ct *naglfarv1.TestClusterTopology, rr *naglfarv1.TestResourceRequest) (requeue bool, err error) {
```

Comment:
![image](https://user-images.githubusercontent.com/59600278/104457781-70024200-55e5-11eb-9526-de285e3b649a.png)
In the code of installing tidb cluster, because the resource may not be initialized, the requeue is right. However, in upgrade, the resource does not need to be initialized. In other cases of failure, it seems that there is no need to try again

Reply:
Just like what you said, why add the requeue bool on method return signature?